### PR TITLE
Add clarity to workflow example doc for generating assignments

### PR DESCRIPTION
**Pull request by willingc**
I've added some detail, edited the prose, and provided some user-friendly format enhancements. Please suggest changes as desired.

This PR relates to Issue #283. 
When I made the changes, I thought ``nbgrader assign`` was part of the issue list. I'm happy to go back and do the others in the issue. I just wanted to get some feedback before I make too many changes :smile:

### CODE REVIEW

**Comment by jhamrick** [comment]
This is awesome! Thanks so much for working on this :-)

The only thing is that you'll need to run `invoke clear_docs` from the root of the repo and recommit, because otherwise there is metadata saved in the notebooks that restricts them to running on a particular kernel (in your case, Python 2) but they need to be able to run on both Python 2 and Python 3. The `clear_docs` task will remove the metadata for you.

File: docs/source/user_guide/03_generating_assignments.ipynb [hunk 1]
```diff
@@ -118,14 +138,42 @@
...
    "source": [
-    "After doing this, there will be a new folder called `release` with the same structure as `source`, but with the actual release version of the files:\n",
+    "### 4.  Inspect the release folder\n",
+    "After doing this, there will be a new folder called `release` with the same structure as `source`. The `release` folder contains the actual release version of the assignment files:\n",
     "\n",
     "* [release/Problem Set 1/Problem 1.ipynb](release/Problem Set 1/Problem 1.ipynb)\n",
     "* [release/Problem Set 1/Problem 2.ipynb](release/Problem Set 1/Problem 2.ipynb)"
    ]
+  },
+  {
```
Comment (jhamrick) [inline docs/source/user_guide/03_generating_assignments.ipynb]: Looks like there is an empty cell at the bottom here.

**Comment by willingc** [comment]
@jhamrick Thanks for the feedback. I realized that I had forgotten the invoke clear_docs when you message came in. Will fix the empty cell and resubmit PR.

**Comment by jhamrick** [comment]
Looks like you need to run `clear_docs` again (you have to do it whenever you save the notebook in the browser, since that's what saves the metadata). I wish there was a less annoying solution than this, but I haven't been able to come up with one yet, sorry :-/

**Comment by willingc** [comment]
@jhamrick I reran clear_docs before the updated PR. I actually think that the conda environment may be preventing the cleaning from happening properly. I will set up a clean environment using pip and see if that helps. I'll also take a look at the Travis build errors to see if I can find a workaround in the conda environment.

**Comment by willingc** [comment]
@jhamrick 5 of the 6 Travis test builds pass green. The 3.4 js build timed out.

Realized that I wasn't actually in the root directory when I executed clear_docs which solved most of the Travis issues. Not sure why the 3.4 js build is timing out. Will restart Travis and see.

**Comment by jhamrick** [comment]
Sometimes the js tests randomly time out (I'm not sure why). I'll restart them and merge once they pass. Thanks!

**Comment by willingc** [comment]
Hmm...I wonder why the js tests time out. Looks green now :)

**Comment by jhamrick** [comment]
Great, thanks!

**Pull request merged**